packages/ckeditor5-font-mapper/src/fontMapping/FontReplacer.ts

```diff
@@ -0,0 +1,174 @@
+    .map((value) => value as Element);
+  return children;
+};
+
+/**
+ * Returns a registered {@link FontMapping} based on the given element.
+ *
+ * A {@link FontMapping} will be looked up by first checking if a font-family exist
+ * on the element. If a font-family exists a font mapping for the font-family will be returned
+ * or undefined if no font mapping exists.
+ * If no font-family exists on the element the inherited font mapping will be returned.
+ *
+ * @param element - the element to inspect
+ * @param inheritedFontMapping - a font mapping inherited from parent hierarchy
+ *    or undefined if no font-family exist in the parent hierarchy
+ * @returns the fontMapping based on the given element or undefined
+ */
+const computeFontMappingForElement = (
+  element: Element,
+  inheritedFontMapping?: FontMapping
+): FontMapping | undefined => {
+  const fontFamily = evaluateFontFamily(element);
+  if (fontFamily) {
+    return getFontMappingForFontFamily(fontFamily);
+  }
+  return inheritedFontMapping;
+};
+
+/**
+ * Returns the value of the element's font-family style property.
+ *
+ * @param element - the element
+ * @returns the font-family or undefined if no font-family style property is set
+ */
+const evaluateFontFamily = (element: Element): string | undefined => {
+  return element.getStyle(FONT_FAMILY_PROPERTY_NAME);
+};
+
+/**
+ * Returns a mapping for the given font-family string from the {@link FontMappingRegistry}.
+ * Only the first font in the font-family string will be taken into account.
+ *
+ * Caution:
+ *
+ * This means that this might no be as precise as it should be. In a real css style
+ * the font-family string can contain fallbacks which means that if the searched <i>fontFamily</i> is
+ * part of the fallbacks won't be respected.
+ *
+ * This might cause problems in some scenarios.
+ * For example font-family style contains two font-families: SomeSpecialFont and the passed font-family.
+ * Now "SomeSpecialFont" is not supported by browsers and the fallback font-family is used to display
+ * the characters in the browser. But the algorithm returns undefined instead of the <i>FontMapping</i>
+ * for the passsed fontFamily.
+ * Reason for this is that we can not say when which style is really applied as it highly depends on the
+ * used browser and installed fonts.
+ *
+ * @param fontFamily - the font family to get a font mapping for.
+ * @returns a {@link FontMapping} or undefined if no such mapping exists.
+ */
+const getFontMappingForFontFamily = (fontFamily: string): FontMapping | undefined => {
+  /*
+   * Splits a string into an array and returns the first element
+   *
+   * Example:
+   * "Symbol, Arial" is converted to ["Symbol", "Arial"]
+   * "Symbol" is converted to ["Symbol"]
+   */
+  const fontName = fontFamily.split(",")[0];
+
+  // Replace quotes, since they are used for some fonts in the font-family string
+  const escapedFontName = fontName.replaceAll('"', "");
```

Comment:
We may want this to an extra method instead, which then can also be unit tested:

```typescript
const firstFontFamily = (fontFamilyStyle: string): string => {
  return fontFamilyStyle
  .split(/\s*(?:^,*|,|$)\s*/)
  .map((s) => s.replace(/^"(.*)"$/, "$1"))
  .filter((s) => !!s)
  [0];
}

const testData = [
  `,Symbol`,
  `Symbol`,
  `Georgia, serif`,
  `"Gill Sans Extrabold", sans-serif`,
  `sans-serif`,
  ` cursive`,
  `system-ui ,serif`,
];

testData.map(firstFontFamily).forEach((f) => console.log(f));
```

See [TypeScript Playground](https://www.typescriptlang.org/play?target=99&jsx=0#code/MYewdgzgLgBAZgSwE7QGLiqghgWwQGwE8YBeGACjg2zyIGUpD8BTALhmiQTAHMBKdp249SAPhgBvALAAoGDCTMoAVyRh41XAUIMmzWfIB0EAA74EUcgHoAOhABU5APysAegBp7AH3deAJHx29lZ8BjCGOFgm5OQQfGIchopmWMDM1q4AROSG9nyZflbuMAUAjJl8oXLhiPhQzEgxcQkAhC1xYQDaAAwAugDcsgC+srKgkLD10AAiWFBYpDCdYQAG7nSEOABGIPhrqxvbu-vVKwDizCBIPAhYxRANCHAn8iuZZwT4MHRYkDAAogAPKBILA7fAAE0y91+EAAtA8uM93KsILCEY9kasYMBVBAEAA3ZgvGArCCEaDMHBw5QIGDuRFPE4DUYyKZQWbzCJRSjINCaWiEPiGKhIf6pAAWMTg8RI4nGEF2zEM+BAPEolUGMiAA).

**Additional Thought:**

We should clearly document, that we only support `font-family` style, and not, for example, `font` style or even some CSS `<style>` section. We may explain this restriction by the clear use-case to hook into paste from office, where we expect that office applications use `font-family`.